src/main/java/com/aws/greengrass/deployment/DeploymentConfigMerger.java

```diff
@@ -147,6 +152,22 @@ private void updateActionForDeployment(Map<String, Object> newConfig, Deployment
             return;
         }
 
+        // As long as the timeout is not 0, we will try to run the connectivity validation
+        int timeout = 0;
+        DeploymentConfigurationValidationPolicy policy = deployment.getDeploymentDocumentObj()
+                .getConfigurationValidationPolicy();
+        if (policy != null) {
+            timeout = Coerce.toInt(policy.timeoutInSeconds());
+        }
+        long timestamp = deployment.getDeploymentDocumentObj().getTimestamp();
+        if (timeout != 0) {
```

Comment:
add a flag like I suggested in the Nucleus config so that we can override the behavior even with a non-zero timeout.

Reply:
I have added a function `isConnectivityValidationEnabled` to the DeviceConfiguration class, that is called inside ConnectivityValidator because the value in deployment config should hold priority over the value in the current config